### PR TITLE
build: Update bazel to 3.4.1 and k/repo-infra to v0.2.0

### PR DESCRIPTION
**Pull request by mikedanese**
build: Update bazel to 3.4.1 and k/repo-infra to v0.2.0
Needed for Golang 1.16 updates: https://github.com/kubernetes/release/issues/1834

/kind cleanup
/sig release

```release-note
NONE
```

### CODE REVIEW

**Comment by mikedanese** [comment]
I don't think we need this until after code freeze, but I'll let @justaugustus say when.

cc @BenTheElder @fejta

**Comment by mikedanese** [comment]
Oops, forgot to push.

**Review by justaugustus — CHANGES_REQUESTED** [review]
Chatted with @mikedanese and we'll want to first:
- bump the bazel version in k/repo-infra
- update any required deps in k/repo-infra
- cut a new k/repo-infra release
- target the new k/repo-infra in this PR
- play whack-a-mole with any failures
- ~curl up in a ball and cry~
- ...
- PROFIT

An additional vector is the go1.16 release that is "imminent", so overall, I think I'd prefer for this to wait until 1.21.

/hold
/milestone v1.21
/area release-eng
cc: @kubernetes/release-engineering

**Comment by mikedanese** [comment]
Are we ready?

/hold cancel

**Comment by mikedanese** [comment]
Sorry, just reread you comment after vacation. We can tie this up with a repo-infra version bump if that's desirable.

**Comment by MushuEE** [comment]
@amwat you might have some insight as to why kubetest is failing on bazel bump

**Comment by mikedanese** [comment]
/retest

Looks like it failed due to flaky networking in the build cluster.

**Comment by BenTheElder** [comment]
looks to me like the bazel sandbox is newly interfering with unix sockets skimming https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/96011/pull-kubernetes-bazel-test/1346729597261058048

**Comment by mikedanese** [comment]
Yup, appears to be something in rules_go. I'll investigate.

**Comment by mikedanese** [comment]
Here's the issue: https://github.com/bazelbuild/rules_go/issues/2776
And a fix: https://github.com/bazelbuild/rules_go/pull/2777

File: staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures/fixtures.go [hunk 1]
```diff
@@ -38,28 +38,43 @@ var (
...
+}
+
+// pkgPath returns the absolute file path to this package's directory. With go
+// test, we can just look at the runtime call stack. However, bazel compiles go
+// binaries with the -trimpath option so the simple approach fails however we
+// can consult environment variables to derive the path.
+//
+// The approach taken here works for both go test and bazel on the assumption
+// that if and only if trimpath is passed, we are running under bazel.
+func pkgPath() (string, error) {
```
Comment (mikedanese) [inline staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures/fixtures.go:59]: This is copied from https://github.com/kubernetes/kubernetes/pull/90872/

**Comment by mikedanese** [comment]
For staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures/fixtures.go change:

/assign @andrewsykim

**Comment by dims** [comment]
/approve

(For code organization / dependency updates only) :)

**Comment by dims** [comment]
/hold

( for @andrewsykim )

**Review by andrewsykim — COMMENTED** [review]
lgtm for `staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures/fixtures.go`

**Comment by justaugustus** [comment]
Awesome work, @mikedanese!!!

**Pull request merged**